### PR TITLE
fix: notification templates for admin reports

### LINKED ISSUE #1809: Add template case for admin.report notifications
Similar to `notification.type === 'admin.sign_up'` in `NotificationCard.vue`, Masto 4.0's notification for new reports should be displayed.

Notification type is `admin.report`, currently an error is shown:
image

### PR DESCRIPTION
**Pull request by kytta**
This PR adds the notification template case for admin.report notifications. Closes #1809

-----
<a href="https://stackblitz.com/~/github.com/kytta/elk/tree/kytta%2Fissue1809">Review PR in StackBlitz Codeflow</a> _Submitted with [StackBlitz Codeflow](https://stackblitz.com/~/github.com/kytta/elk/tree/kytta%2Fissue1809)._

### CODE REVIEW

**Comment by kytta** [comment]
This is not ready, I need some help from the maintainers.

First, what is the best way to return the Route to the actual Mastodon instance? Is there a Router entity that would achieve this for me?

Second, I have problems with `masto`'s `Notification` type, as it does not include the optional `report` attribute ([which is present in the actual entity](https://docs.joinmastodon.org/entities/Notification/))

Third, I have only added the English localization for now. In other languages (like German), the translation would require two words (_$USER hat $TARGET_USER gemeldet_). Is there a way to author i18n messages with templates? I haven't got much experience with the library.

Apart from that, I think this is enough for the issue to be fixed.

**Comment by kytta** [comment]
I'm happy with the end result so far. The only problem I'm having right now is the spacing between "reported" and the target account name:

image

No idea how to fix it :/

File: components/notification/NotificationCard.vue [hunk 1]
```diff
@@ -39,6 +39,23 @@ const { notification } = defineProps<{
...
+        <div flex p3 items-center bg-shaded>
+          <div i-ri:flag-fill me-1 color-purple />
+          <i18n-t keypath="notification.reported">
+            <AccountDisplayName
+              :account="notification.account"
+              text-purple me-1 font-bold line-clamp-1 ws-pre-wrap break-all
+            />
+            <AccountDisplayName
+              :account="notification.report?.targetAccount!"
+              text-purple me-1 font-bold line-clamp-1 ws-pre-wrap break-all
```
Comment (patak-dev) [inline components/notification/NotificationCard.vue:53]: This fixes the spacing issue
```suggestion
              text-purple ms-1 font-bold line-clamp-1 ws-pre-wrap break-all
```

**Review by patak-dev — APPROVED** [review]

**Comment by patak-dev** [comment]
Thanks for the PR!

File: composables/masto/routes.ts [hunk 2]
```diff
@@ -29,6 +29,10 @@ export function getAccountFollowersRoute(account: mastodon.v1.Account) {
   })
 }
 
+export function getReportRoute(id: string | number) {
+  return `https://${currentUser.value?.server}/admin/reports/${id}`
```
Comment (danielroe) [inline composables/masto/routes.ts:33]:
```suggestion
  return `https://${currentUser.value?.server}/admin/reports/${encodeURIComponent(id)}`
```

**Pull request merged**